internal/pkg/service/templates/api/service/mapper.go

```diff
@@ -579,3 +583,18 @@ func instanceDetails(ctx context.Context, d dependencies.ProjectRequestScope, in
 
 	return tmplResponse, versionResponse
 }
+
+func hasRequirements(tmpl repository.TemplateRecord, d dependencies.ProjectRequestScope) bool {
+	if tmpl.Requirements.Backends != nil && !tmpl.HasBackend(d.ProjectBackends()) {
+		return false
+	}
+
+	if tmpl.Requirements.Components != nil && !tmpl.CheckComponents(d.Components()) {
+		return false
+	}
+
+	if tmpl.Requirements.Features != nil && !tmpl.CheckFeatures(d.ProjectFeatures()) {
+		return false
+	}
+	return true
```

Comment:
Please, remove these `!= nil` checks, for components and features, they are not needed.

About the backends, see comment in the `CheckFeature` method.